doc/dev.md

```diff
@@ -0,0 +1,15 @@
+# Developer documentation
+
+## Warnings
+
+`parameterized-utils` is a research-grade codebase, so correctness in all cases is not *always* the highest priority.
```

Comment:
Thank you for this writeup!

I think it would be good to define "correctness" here, since I think we would all agree that correctness in terms of proper output is a priority, but what we are talking about here is compilation concerns and so is probably better referred to as "code hygene", or with a good reference to what type of "correctness" is being discussed.

Reply:
Ah, this is a helpful point of clarification - I actually *did* mean correctness in terms of proper output, as `-Wdefault` includes warnings such as `-Wmissing-methods` and `-Woverlapping-patterns`, and I would say that the presence of such a warning often indicates a "correctness in terms of proper output"-type bug.

However, I agree that some members of `-Wdefault` do *not* concern this type of correctness, such as `-Wtabs`.

I will make this distinction more clear.

Reply:
Pushed another commit that highlights correctness as a major concern, and mentions that the purpose of `-Wdefault` is to reduce bugs, though also includes a footnote on ways in which that is not quite true in practice.

Reply:
Looks good.  And yes, I frequently turn on the warnings you mentioned in other projects because of the implications, whilst not usually going as far as  `-Werror -Wdefault`.